queries.sql

```diff
@@ -159,4 +159,78 @@ FROM owners o
 LEFT JOIN animals a ON o.id = a.owner_id
 GROUP BY o.full_name
 ORDER BY num_of_animals_owned DESC
-LIMIT 1;
\ No newline at end of file
+LIMIT 1;
+
+
+--- 1. Who was the last animal seen by William Tatcher?
+SELECT a.name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'William Tatcher'
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+--- 2. How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT a.id) AS num_of_animals_seen
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'Stephanie Mendez';
+
+--- 3. List all vets and their specialties, including vets with no specialties.
+SELECT v.name AS vet_name, COALESCE(s.name, 'No Specialization') AS specialty
+FROM vets v
+LEFT JOIN specializations sp ON v.id = sp.vet_id
+LEFT JOIN species s ON sp.species_id = s.id;
+
+
+--- 4. List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'Stephanie Mendez' AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+--- 5. What animal has the most visits to vets?
+SELECT a.name AS animal_name, COUNT(a.id) AS num_of_visits
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+GROUP BY a.name
+ORDER BY num_of_visits DESC
+LIMIT 1;
+
+--- 6. Who was Maisy Smith's first visit?
+SELECT a.name AS animal_name, vt.name AS vet_name, MIN(v.visit_date) AS first_visit_date
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY a.name, vt.name;
+

```

Comment:
- [ ] This query us multiple records but we want only one 🤔 It would be nice if you return only one record as required that `Who was Maisy Smith's first visit?` 😊